include/seqan3/search/algorithm/detail/search_trivial.hpp

```diff
@@ -32,10 +38,11 @@ namespace seqan3::detail
  * \tparam cursor_t      Must model seqan3::FmIndexCursor.
  * \tparam query_t       Must model std::ranges::InputRange over the index's alphabet.
  * \tparam delegate_t    Takes `index::cursor_type` as argument.
- * \param[in] cur        Cursor of atring index built on the text that will be searched.
+ * \param[in] cur        Cursor of a string index built on the text that will be searched.
  * \param[in] query      Query sequence to be searched with the cursor.
  * \param[in] query_pos  Position in the query sequence indicating the prefix that has already been searched.
  * \param[in] error_left Number of errors left for matching the remaining suffix of the query sequence.
+ * \param[in] memory     Previous enumerate error or match.
```

Comment:
I'd suggest calling the enum something like `previous_outcome`, `previous_result` or something like that since `last` is a bit ambiguous and `error` does not encompass `match`.
```suggestion
 * \param[in] prev_result     Previous scenario of search(), i.e. error or match.
```

Or something like that.

Reply:
I agree. I also changed the enum class name. But, I am still not fully satisfied.
Maybe enum_type is better, but that would also be confusing since it is a enum class. 
last_error -> error_type
memory -> prev_enumeration

Reply:
I'd move away from naming it something with `enum` since that's how it's implemented and not really what it means. 
I guess we can go with `error_type` and `prev_error` for now?